multiversx_sdk_core/transaction.py

```diff
@@ -1,156 +1,119 @@
-        gas_price: Optional[IGasPrice] = None,
-        nonce: Optional[INonce] = 0,
-        value: Optional[ITransactionValue] = None,
-        data: Optional[ITransactionPayload] = None,
-        version: Optional[ITransactionVersion] = None,
-        options: Optional[ITransactionOptions] = None,
-        guardian: Optional[IAddress] = None
-    ) -> None:
-        self.chainID: IChainID = chain_id
-        self.sender: IAddress = sender
-        self.receiver: IAddress = receiver
-
-        self.sender_username: str = sender_username or ""
-        self.receiver_username: str = receiver_username or ""
-
-        self.guardian: Optional[IAddress] = guardian
-
-        self.gas_limit: IGasLimit = gas_limit
-        self.gas_price: IGasPrice = gas_price or TRANSACTION_MIN_GAS_PRICE
-
-        self.nonce: INonce = nonce or 0
-        self.value: ITransactionValue = value or 0
-        self.data: ITransactionPayload = data or TransactionPayload.empty()
-
-        self.version: ITransactionVersion = version or TRANSACTION_VERSION_DEFAULT
-        self.options: ITransactionOptions = options or TRANSACTION_OPTIONS_DEFAULT
-
-        self.signature: ISignature = bytes()
-        self.guardian_signature: ISignature = bytes()
-
-    def serialize_for_signing(self) -> bytes:
-        dictionary = self.to_dictionary(with_signature=False)
+    def __init__(self,
+                 sender: str,
+                 receiver: str,
+                 gas_limit: int,
+                 chain_id: str,
+                 nonce: Optional[int] = None,
+                 amount: Optional[int] = None,
+                 sender_username: Optional[str] = None,
+                 receiver_username: Optional[str] = None,
+                 gas_price: Optional[int] = None,
+                 data: Optional[bytes] = None,
+                 version: Optional[int] = None,
+                 options: Optional[int] = None,
+                 guardian: Optional[str] = None,
+                 signature: Optional[bytes] = None,
+                 guardian_signature: Optional[bytes] = None
+                 ) -> None:
+        self.chainID = chain_id
+        self.sender = sender
+        self.receiver = receiver
+        self.gas_limit = gas_limit
+
+        self.nonce = nonce or 0
+        self.amount = amount or 0
+        self.data = data or bytes()
+        self.signature = signature or bytes()
+
+        self.sender_username = sender_username or ""
+        self.receiver_username = receiver_username or ""
+
+        self.gas_price = gas_price or TRANSACTION_MIN_GAS_PRICE
+        self.version = version or TRANSACTION_VERSION_DEFAULT
+        self.options = options or TRANSACTION_OPTIONS_DEFAULT
+
+        self.guardian = guardian or ""
+        self.guardian_signature = guardian_signature or bytes()
+
+
+class TransactionComputer:
+    def __init__(self) -> None:
+        pass
+
+    def compute_transaction_fee(self, transaction: Transaction, network_config: INetworkConfig) -> int:
```

Comment:
Should be unit tested (in this PR or the next one).

Reply:
Added the same unit tests as in `js-core`.